docs/fsharp/language-reference/fsharp-interactive-options.md

```diff
@@ -62,6 +57,106 @@ Where lists appear in F# Interactive option arguments, list elements are separat
+
+F# Interactive (`dotnet fsi`) uses an extended version of [structured plain text formatting](plaintext-formatting.md) to
+report values.
+
+1. All features of `%A` plain text formatting are supported, and some are additionally customizable.
+
+2. Printing is colorized if colors are supported by the output console.
+
+3. A limit is placed on the length of strings shown, unless you explicitly evaluate that string.
+
+4. A set of user-definable settings are available via the `fsi` object.
+
+The available settings to customize plain text printing for reported values are:
+
+```fsharp
+open System.Globalization
+
+fsi.FormatProvider <- CultureInfo("de-DE")  // control the default culture for primitives
+
+fsi.PrintWidth <- 120        // Control the width used for structured printing
+
+fsi.PrintDepth <- 10         // Control the maximum depth of nested printing
+
+fsi.PrintLength <- 10        // Control the length of lists and arrays
+
+fsi.PrintSize <- 100         // Control the maximum overall object count
+
+fsi.ShowProperties <- false  // Control whether properties of .NET objects are shown by default
+
+fsi.ShowIEnumerable <- false // Control whether sequence values are expanded by default
+
+fsi.ShowDeclarationValues <- false // Control whether values are shown for declaration outputs
+```
+
+### Customizing F# Interactive structured printing with `AddPrinter` and `AddPrintTransformer`
+
+Printing in F# Interactive outputs can be customized by using `fsi.AddPrinter` and `fsi.AddPrintTransformer`.
+The first gives text to replace the printing of an object, the second returns a surrogate object to display
+instead. For example, in F# Interactive
+
+```fsharp
+open System
+
+fsi.AddPrinter<DateTime>(fun dt -> dt.ToString("s"))
+
+type DateAndLabel =
+  { Date : DateTime
+    Label  : string  }
+
+let newYearsDay1999 =
+  { Date = DateTime(1999, 1, 1)
+    Label = "New Year" }
+```
+
+gives output that uses the specific date time format:
+
+```console
+type DateAndLabel =
+  { Date: DateTime
+    Label: string }
+val newYearsDay1999 : DateAndLabel = { Date = 1999-01-01T00:00:00
+                                       Label = "New Year" }
+```
+
+`fsi.AddPrintTransformer` can be used to give a surrogate object for printing:
+
+```fsharp
+type MyList(values: int list) =
+    member x.Values = values
+
+fsi.AddPrintTransformer(fun (x:MyList) -> box x.Values)
+
+let x = MyList([1..10])
+```
+
+gives
+
+```console
+val x : MyList = [1; 2; 3; 4; 5; 6; 7; 8; 9; 10]
+```
+
+If the transformer function passed to `fsi,AddPrintTransformer` returns `null` then the print transformer is ignored.
+This can be used to filter any input value by starting with type `obj`.  For example,
+
+```fsharp
+fsi.AddPrintTransformer(fun (x:obj) ->
+    match x with
+    | :? string as s when s = "beep" -> box ["quack"; "quack"; "quack"]
+    | _ -> null)
+
+let y = "beep"
+```
+
+gives
+
+```console
+val y : string = ["quack"; "quack"; "quack"]
+```
+
 ## Related Topics
```

Comment:
I won't add too much suggestions on the same thing, but the idea is to use sentence-style casing, I think that's what the current style-guide says.

```suggestion
## Related topics
```